modules/integration/src/test/java/io/ballerina/messaging/broker/integration/standalone/amqp/NegativeQueueDeleteTest.java

```diff
@@ -68,6 +68,21 @@ public void testDeleteQueueWithMessages() throws Exception {
         channel.queueDelete(queueWithMessages, false, true);
     }
 
+    @Test(description = "Test queue auto delete", expectedExceptions = IOException.class)
+    public void testQueueAutoDelete() throws Exception {
+        Channel channel = amqpConnection.createChannel();
+
+        String queueName = "PositiveQueueDeleteTestTestQueueAutoDelete";
+        channel.queueDeclare(queueName, false, false, true, null);
+        String consumerTag = channel.basicConsume(queueName, new DefaultConsumer(channel));
+
+        channel.basicCancel(consumerTag);
+
+        // This should throw an exception since the queue is auto deleted
+        channel.queueDeclarePassive(queueName);
```

Comment:
Purpose of doing that?

Reply:
Then it is verified that the queue is removed. Not getting an exception does not necessarily mean that the queue is removed. If we have a util method to check on queue existence, it will be useful and reusable.

Reply:
Resource declare with passive bit set is the AMQP way of checking the resource existence. So IMO this does the same thing.